text/0031-designated-operators.md

```diff
@@ -0,0 +1,93 @@
+
+For safety, consensus messages from a validator signed with validator's own private key ought to be ignored in an era where this validator has a designated operator.
+
+### Cases
+
+Genesis validators - operator specified at platform launch
+
+Entrant (no history as bidder or validator) - optionally supplies operator public key in `add_bid`
+
+Existing bid - changes bid data (by using `add_bid`, similar to adjusting bid token amount) to include an operator public key; platform will expect this operator to sign messages on validator's behalf in a future era where this bid wins
+
+### Typical timeline
+
+Bid entered by a new entrant B, no operator provided
+
+... (some blocks are added to the linear chain)
+
+Auction for era N + `auction_delay` takes place, B wins
+...
+
+Era N + `auction_delay` starts, B is a validator, signing own consensus messages with own private key
+
+During era N + `auction_delay`, B calls `add_bid` to supply an operator public key (and possibly add/reduce number of tokens in the bid)
+
+Auction for era N + `auction_delay` + `auction_delay` takes place, B wins again
+
+...
+
+Era N + `auction_delay` + `auction_delay` starts, B is a validator, but consensus messages are produced and signed by the operator, consensus messages signed by B are ignored
+
+### Restrictions on designation
+
+In order to simplify reasoning about validator-operator pair behavior and to prevent potential introduction of equivocation or reward bugs, operator designation should be checked against the following:
+
+1. Validator public keys slated to participate in future eras, other than that of the designator.
+2. Validator public keys that are participating in current era or have participated in past eras, up to the unbonding wait period, other than that of the designator.
+3. Bidder public keys for the upcoming auction, other than that of the designator.
+3. Operator public keys corresponding to validator public keys in categories 1 through 3.
+
+Should a designating deploy specify a "duplicate" public key contained in the categories 1 through 4, the deploy ought to be rejected. Note that we allow for the possibility of designating oneself as the operator, with no changes to current node or consensus behavior.
```

Comment:
Execution of the transaction has to fail (versus deploy being rejected)

Reply:
Will change